src/azure-cli/azure/cli/command_modules/synapse/commands.py

```diff
@@ -139,8 +139,12 @@ def get_custom_sdk(custom_module, client_factory):
         operations_tmpl='azure.synapse.spark.operations#SparkBatchOperations.{}',
         client_factory=None)
 
-    synapse_accesscontrol_sdk = CliCommandType(
-        operations_tmpl='azure.synapse.accesscontrol.operations#AccessControlClientOperationsMixin.{}',
+    synapse_role_assignment_sdk = CliCommandType(
+        operations_tmpl='azure.synapse.accesscontrol.operations#RoleAssignmentsOperations.{}',
+        client_factory=None)
```

Comment:
why your client factory is None?

Reply:
That is a dataplane command, so we need to create client using workspace name which is provided by user from parameter. So the client will be created when CLI received parameter instead of initialization at here. 

For the management commands, client factory is not None.

Reply:
Actually it doesn't matter because client factory will take effect during command execution where user parameter is already loaded. 
For `-h`, we are using operation_tmpl to get parameter list. 
You could see `storage blob` related command, which is data plane command and need user input to create a client factory. 
It is nice to have if you can refactor your design but as we are close to release date. I could accept current code now.